site/documentation/content/concepts/device-identity.md

```diff
@@ -147,3 +147,34 @@ After having verified the client certificate using the trust anchor(s), the prot
 the client certificate's *subject DN* and invokes the Credentials API's
 [*get Credentials*]({{< relref "/api/credentials#get-credentials" >}}) operation in order to retrieve the
 [*x509-cert* type credentials]({{< relref "/api/credentials#x509-certificate" >}}) that are on record for the device.
+
+### JSON Web Token based Authentication
+
+The MQTT protocol adapter supports authentication of devices with a JSON Web Token (JWT) based mechanism. In this case,
+the protocol adapter verifies the JWT by using the public key / certificate that is on record for the device in the
+device registry. Further more does the protocol adapter check if the JWT is valid. To be valid the JWT has to provide
+the following information in its header and payload.
+
+The JWT header has to provide two fields that indicate the type of token (`typ`) and the signing algorithm (`alg`).
+Both fields are mandatory and the type has to be `JWT`. For the signing algorithm `RS256` and `ES256` are supported.
+The algorithm specified in the header must match at least one of the
+[*asymmetric-key* type credentials]({{< relref "/api/credentials#asymmetric-key" >}}) registered for the device.
+
+The JWT payload must at least contain the claims `iat` ("issued at") and `exp` ("expiration time") with values provided
+in [Unix time](https://en.wikipedia.org/wiki/Unix_time). The `iat` claim marks the timestamp the token was created at
+and will also represent the start of its validity period. The `nbf` ("not before") claim is therefore not required and
+will be ignored. The `exp` claim signifies the point in time after which the token becomes invalid.
+
+During connection establishment the device presents a client identifier, a username and a password to the protocol
+adapter. The JWT has to be provided in the password field. The username will be ignored; however, some MQTT client
+libraries will not send the password unless the username is specified. The client identifier must provide the
+*tenant-id* and *auth-id* in the following format: */*tenant-id*/device/*auth-id*. For example, a device that belongs
```

Comment:
does the leading `*` character indicate a glob pattern? For example, would the following client identifiers be valid for a device of tenant `my-tenant` with auth-id `my-auth-id`:
1. `some/leading/segments/my-tenant/device/my-auth-id`
2. `/my-tenant/device/my-auth-id`
3. `/leading-slash/my-tenant/device/my-auth-id`

I also wonder, if we really want to determine the tenant and auth id from the MQTT client identifier instead of from claims in the JWT. If we use the client identifier, then this will also only work for the MQTT adapter whereas we could support this mechanism in all adapters if we use the JWT for conveying the information ...

Reply:
Yes, all of your examples would work. The idea behind this approach is to keep the migration effort for people migration from Google IoT Core as low as possible. Google IoT Core expects the client identifier to be of the following format: projects/PROJECT_ID/locations/REGION/registries/REGISTRY_ID/devices/DEVICE_ID. So with the currently chosen approach these client identifiers do not have to be changed.

Reply:
is `/devices/` as in your example or `/device/` as in the documentation?

Reply:
Will retrieving the tenant and auth ID from the MQTT client identifier allow the firmware running on devices to remain completely unchanged?

Reply:
Sadly not completely, since the device needs a new targeting URL (URL of MQTT adapter) and a slight modification of the MQTT topics. But the big benefit would be that the authentication (e.g. creation of JWTs, etc.) could stay exactly the same.

Reply:
> is it `/devices/` as in your example or `/device/` as in the documentation?

`/devices/ ` is correct.

Reply:
> > is it `/devices/` as in your example or `/device/` as in the documentation?
> 
> `/devices/ ` is correct.

Technically speaking both are valid, as this is not validated. But I will see to it that it is at least consistent.